pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/PublishDraftVersion.java

```diff
@@ -85,10 +86,10 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
         } else {
             // Draft becomes the new released version
             ModuleLocale moduleLocale = module.getModuleLocale(locale);
-            moduleLocale.released().set( moduleLocale.draft().get() );
-            moduleLocale.draft().set( null );
+            moduleLocale.archiveReleasedVersion();
+            ModuleVersion releasedVersion = moduleLocale.releaseDraftVersion();
```

Comment:
Yeah, perfect followup from my previous comment - it doesn't make sense for the developer to need to call both 'archive' and 'release' methods - the release process should imply any necessary archiving.

Reply:
We still need the `archiveReleasedVersion` to be separate as we use it in the unpublish operation.
But I could create more "business level" methods which just `publishDraft` and `recallReleased` 
Thoughts on that approach?